include/accessor.h

```diff
@@ -302,17 +402,53 @@ class accessor<DataT, Dims, Mode, cl::sycl::access::target::host_buffer>
 };
 
 namespace detail {
-
 	template <typename DataT, int Dims, cl::sycl::access_mode Mode, typename... Args>
-	accessor<DataT, Dims, Mode, cl::sycl::access::target::global_buffer> make_device_accessor(Args&&... args) {
+	accessor<DataT, Dims, Mode, cl::sycl::target::global_buffer> make_device_accessor(Args&&... args) {
 		return {std::forward<Args>(args)...};
 	}
 
 	template <typename DataT, int Dims, cl::sycl::access_mode Mode, typename... Args>
-	accessor<DataT, Dims, Mode, cl::sycl::access::target::host_buffer> make_host_accessor(Args&&... args) {
+	accessor<DataT, Dims, Mode, cl::sycl::target::host_buffer> make_host_accessor(Args&&... args) {
 		return {std::forward<Args>(args)...};
 	}
 
+	template <typename TagT>
+	constexpr cl::sycl::access_mode deduce_access_mode() {
+		if constexpr(std::is_same_v<const TagT, decltype(cl::sycl::read_only)> || //
```

Comment:
This (and all below) doesn't need to be `constexpr`. Not sure if we'd want to adopt a consistent policy regarding this... cc @fknorr.